### PR TITLE
chore: split installation and upgrade process

### PR DESCRIPTION
**Pull request by alejandroEsc**
chore: split installation and upgrade process

**What type of PR is this?**

Chore

**What this PR does/ why we need it**:

PR resolves issues where current CI will downgrade instead of fresh installing causing stateful addons to fail if api changes occur and data stored in a PV not valid in previous version.

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Checklist**

* [ ] The commit message explains the changes and why are needed.
* [ ] The code builds and passes lint/style checks locally.
* [ ] The relevant subset of integration tests pass locally.
* [ ] The core changes are covered by tests.
* [ ] The documentation is updated where needed.

### CODE REVIEW

**Comment by gracedo** [comment]
I think I found the issue causing these tests to fail, commented here https://github.com/mesosphere/kubernetes-base-addons/pull/809/files#r564745622

File: test/groups.yaml [hunk 1]
```diff
@@ -79,9 +82,6 @@ prometheus:
     - "prometheus"
     - "prometheusadapter"
     - "opsportal"
-    - "cert-manager"
```
Comment (alejandroEsc) [inline test/groups.yaml]: think you meant to only copy these right?

File: test/addons_test.go [hunk 2]
```diff
@@ -52,7 +52,7 @@ const (
 	kubeaddonsControllerNamespace = "kubeaddons"
 	kubeaddonsControllerPodPrefix = "kubeaddons-controller-manager-"
 
-	prometheusGroupName = "prometheus"
```
Comment (alejandroEsc) [inline test/addons_test.go]: Why remove the prometheus group name?

**Comment by alejandroEsc** [comment]
was able to fix the issues why this was failing, running another test with a few more changes and will rebase to squash commits

**Comment by alejandroEsc** [comment]
ok passed again. So looking really good. Next going to remove comments for install only.

**Comment by alejandroEsc** [comment]
Installation and upgrade both passed. Working on removing testing charts (from Elastic search and Kibana PR). Now removing these charts, leaving all the changes we want to push on master. If green then we are ready.

File: test/dispatch-ci.sh [hunk 3]
```diff
@@ -8,7 +8,7 @@ set -euxo pipefail
 cd "$(dirname "$0")"
 
 echo "Setup Konvoy"
-source ./scripts/setup-konvoy.sh v1.5.0
+source ./scripts/setup-konvoy.sh v1.6.1
```
Comment (s12chung) [inline test/dispatch-ci.sh:11]: thanks

**Review by s12chung — APPROVED** [review]

File: test/addons_test.go [hunk 4]
```diff
@@ -259,9 +328,38 @@ func cleanupNodeVolumes(numberVolumes int, nodePrefix string, node *v1alpha4.Nod
 	return nil
 }
 
-func testgroup(t *testing.T, groupname string, version string, jobs ...clusterTestJob) error {
+func testgroup(t *testing.T, groupName string, version string, jobs ...clusterTestJob) error {
```
Comment (gracedo) [inline test/addons_test.go:331]: 🙌

**Review by gracedo — APPROVED** [review]
🥳

**Review by joejulian — CHANGES_REQUESTED** [review]

File: test/addons_test.go [hunk 5]
```diff
@@ -564,33 +767,42 @@ kubeEtcd:
...
+		provisioner, ok := os.LookupEnv("KBA_PROVISIONER")
+		if ok && provisioner != "" {
+			return testcluster.NewClusterFromKubeConfig(provisioner, kubeConfig)
+		} else {
+			return nil, fmt.Errorf("required provisioner key %q, not found or empty", "KBA_PROVISIONER")
+		}
+
+		return testcluster.NewClusterFromKubeConfig(provisionerAWS, kubeConfig)
+	} else {
+		path, _ = os.Getwd()
```
[the remark this answers is not in the document]
Comment (alejandroEsc) [reply test/addons_test.go]: i just want this to pass for now. This is holding back the work on ES and i rather not spend a few more hours debugging additional issues atm.

**Review by joejulian — APPROVED** [review]

**Comment by joejulian** [comment]
@mesosphere-mergebot backport release/3.3

**Pull request merged**